src/typing/type_annotation.ml

```diff
@@ -852,9 +852,12 @@ let rec convert cx tparams_map = Ast.Type.(function
     )
 
   | "Function$Prototype$Apply" ->
-    check_type_arg_arity cx loc t_ast targs 0 (fun () ->
-      let reason = mk_reason RFunctionType loc in
-      reconstruct_ast (FunProtoApplyT reason) None
+    let reason = mk_reason RFunctionType loc in
+    (match convert_type_params () with
```

Comment:
i think this is the only place we'd allow a sometimes-parameterized generic. the most similar concept is having a default, but in that case you'd have to use an empty `<>` (https://flow.org/en/docs/types/generics/#toc-adding-defaults-to-parameterized-generics).

this feels like a new behavior that we should consider carefully. also, we should consider how it interacts with annotating `this`, because it might be closer to `Function$Prototype$Apply<this=T>` or whatever syntax we end up with for that -- we've gone back and forth about annotating `this` for functions as either `function f(this: T, ...)` or `function f<this = T>(...)` (cc @gabelevi )

Reply:
I like first one because it mirrors TypeScript

Reply:
How can I make this support `<>`?

Reply:
@mroch you can mimic second one with first one
`type F<T> = (this: T) => void`